src/rust/engine/src/externs/address.rs

```diff
@@ -153,12 +156,19 @@ impl AddressInput {
         _cls: &Bound<'_, PyType>,
         spec: &str,
         description_of_origin: &str,
-        relative_to: Option<&str>,
```

Comment:
These changes from `&str` to `String` are necessiated by the `&str` lifetime when extracted no longer lasting as long as the `'py` GIL lifetime. See [the discussion in the PyO3 migration guide](https://pyo3.rs/main/migration#deactivating-the-gil-refs-feature) for how to deal with this change.

The guide advises to use the new `PyBackedStr` instead of `&str` but in this method the `&str` lifetime was complex enough to warrant just extracting an owned `String`.

Reply:
I'm assuming we don't expect the allocation overhead to be user-noticeable?

Reply:
I didn't get far enough to consider that since I plan to find a way to use references again. The code was encountering "fun" error messages as I tried to sort out the lifetimes for the references. Easier to do this and get the upgrade done first and then come back to this function with PyO3 v0.22 in place and have a focused PR to do it the right way.